spec/duration.html

```diff
@@ -105,9 +105,7 @@ <h1>Temporal.Duration.compare ( _one_, _two_ [ , _options_ ] )</h1>
         1. Let _calendarRec_ be *undefined*.
         1. If _zonedRelativeTo_ is not *undefined* or _plainRelativeTo_ is not *undefined*, then
           1. If _zonedRelativeTo_ is not *undefined*, let _calendar_ be _zonedRelativeTo_.[[Calendar]]; else let _calendar_ be _plainRelativeTo_.[[Calendar]].
-          1. Set _calendarRec_ to ! CreateCalendarMethodsRecord(_calendar_, « »).
-          1. If _calendarUnitsPresent_ is *true*, then
-            1. Perform ? CalendarMethodsRecordLookup(_calendarRec_, ~dateAdd~).
+          1. Set _calendarRec_ to ? CreateCalendarMethodsRecord(_calendar_, « ~dateAdd~ »).
```

Comment:
I wonder if the four instances of this _`If _zonedRelativeTo_ is not *undefined* or _plainRelativeTo_ is not *undefined*, … Set _calendarRec_ to ? CreateCalendarMethodsRecord(_calendar_, « … »)`_ pattern justify introducing a new operation.
```suggestion
        1. Let _calendarRec_ be CreateCalendarMethodsRecordFromField(« _zonedRelativeTo_, _plainRelativeTo_ », « ~dateAdd~ »).
```

Reply:
Thanks for the suggestion. Added MaybeCreateCalendarMethodsRecordFromRelativeTo.